### PR TITLE
Move ActivatorUtilities to DI

### PR DESCRIPTION
**Pull request by pakrym**
https://github.com/aspnet/Common/issues/245#event-1180973596

### CODE REVIEW

**Comment by pranavkm** [comment]
@Tratcher \ @davidfowl  could we modify `Http.Abstractions` to reference `DI.Abstractions`? DI's copy of `ActivatorUtils` is exposed via DI.Abstractions and `Http.Abstractions` is the only project that references the sources version of `ActivatorUtils` today.

**Comment by davidfowl** [comment]
@pranavkm nah. Let's not take a new dependency

File: shared/Microsoft.Extensions.ActivatorUtilities.Sources/ActivatorUtilities.cs [hunk 1]
```diff
@@ -0,0 +1,305 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Reflection;
+using System.Runtime.ExceptionServices;
+
+namespace Microsoft.Extensions.Internal
```
Comment (pranavkm) [inline shared/Microsoft.Extensions.ActivatorUtilities.Sources/ActivatorUtilities.cs]: Could we use `ifdefs` to replace https://github.com/aspnet/DependencyInjection/blob/dev/src/DI.Abstractions/DIActivatorUtilities.cs? Something like https://github.com/aspnet/Common/blob/dev/shared/Microsoft.Extensions.WebEncoders.Sources/WebEncoders.cs#L18?

File: shared/Microsoft.Extensions.ActivatorUtilities.Sources/ActivatorUtilities.cs [hunk 2]
```diff
@@ -0,0 +1,305 @@
...
+{
+    internal delegate object ObjectFactory(IServiceProvider serviceProvider, object[] arguments);
+
+    // Do not take a dependency on this class unless you are explicitly trying to avoid taking a
```
Comment (pranavkm) [inline shared/Microsoft.Extensions.ActivatorUtilities.Sources/ActivatorUtilities.cs]: This comment is not super useful. I don't think anyone actually reads these messages.

File: shared/Microsoft.Extensions.ActivatorUtilities.Sources/sharedsources.props [hunk 3]
```diff
@@ -0,0 +1,8 @@
+<Project>
+  <ItemGroup>
+    <Compile Include="$(NuspecBasePath)\..\Microsoft.Extensions.ParameterDefaultValue.Sources\ParameterDefaultValue.cs">
```
Comment (pranavkm) [inline shared/Microsoft.Extensions.ActivatorUtilities.Sources/sharedsources.props]: Can a source package reference another source package? Could we potentially leave `ParameterDefaultValue` in common and use a package reference?

Comment (pakrym) [inline shared/Microsoft.Extensions.ActivatorUtilities.Sources/sharedsources.props]: I tried hard but wasn't able to make it work. My other though was just to leave them slip and reference both where we reference ActivatorUtils

Comment (pranavkm) [inline shared/Microsoft.Extensions.ActivatorUtilities.Sources/sharedsources.props]: Fair enough

**Comment by pakrym** [comment]
:up::date: `#ifdef`ed ActivatorUtils

**Comment by pranavkm** [comment]
:shipit:

**Pull request merged**